dashboard/src/components/AppUpgrade/AppUpgrade.tsx

```diff
@@ -39,26 +40,71 @@ interface IAppUpgradeProps {
   fetchChartVersions: (id: string) => Promise<IChartVersion[]>;
   getAppWithUpdateInfo: (releaseName: string, namespace: string) => void;
   getChartVersion: (id: string, chartVersion: string) => void;
+  getDeployedChartVersion: (id: string, chartVersion: string) => void;
   push: (location: string) => RouterAction;
   goBack: () => RouterAction;
   fetchRepositories: () => void;
-  enableBasicForm: boolean;
 }
 
 interface IAppUpgradeState {
-  selectRepoForm: JSX.Element;
+  repo?: IAppRepository;
 }
 
 class AppUpgrade extends React.Component<IAppUpgradeProps, IAppUpgradeState> {
+  public state: IAppUpgradeState = {};
+
   public componentDidMount() {
     const { releaseName, getAppWithUpdateInfo, namespace, fetchRepositories } = this.props;
     getAppWithUpdateInfo(releaseName, namespace);
     fetchRepositories();
   }
 
+  public componentWillReceiveProps(nextProps: IAppUpgradeProps) {
```

Comment:
We seem to keep adding more `componentWillReceiveProps` when we should be removing them as we go, not adding?

See https://reactjs.org/docs/react-component.html#updating

The issue here is that we're adding more code that is dependent on detecting a change in the prop values, rather than working towards the functional `getDerivedStateFromProps`. Even if we can't use that here (it's added in 16.3), let's not add code which we're going to have to remove/update in the future? We should be considering the reasons why react is removing `componentWillReceiveProps` (ie. so that people won't be deriving state based on existing state or old prop values).